image-rs/docs/pull_image_with_nydus.md

```diff
@@ -0,0 +1,95 @@
+# Pull image with Nydus
+
+## Prepare the environments
+
+### 1. Install All Nydus Binaries
+Nydus binaries are required to convert container images or directories to (encrypted) Nydus images. Follow the instructions to install Nydus.
+
+Get `nydus-image`, `nydusd`, `nydusify`, `nydusctl` and `nydus-overlayfs` binaries from [release](https://github.com/dragonflyoss/image-service/releases) page.
+
+```shell
+sudo install -D -m 755 nydusd nydus-image nydusify nydusctl nydus-overlayfs /usr/bin
+```
+
+### 2. Deploy and Configure CoCo Key Broker System cluster
+A tenant-side CoCo Key Broker System cluster includes:
+* Key Broker Service (KBS): Brokering service for confidential resources.
+* Attestation Service (AS): Verifier for remote attestation.
+* Reference Value Provicer Service (RVPS): Provides reference values for AS.
+* CoCo Keyprovider: Component to encrypt the images following ocicrypt spec.
+
+We will use the Key Broker System cluster to encrypt the image.
+Follow the [instructions](https://github.com/confidential-containers/documentation/blob/main/quickstart.md#deploy-and-configure-tenant-side-coco-key-broker-system-cluster) to prepare CoCo Key Broker System.
+
+----
+
+## Generate encrypted Nydus image
+Use `nysusify` and `nydus-image` to convert OCI container images to encrypted nydus image with KBS cluster.
+Use busybox:latest for example:
+
+```shell
+cat > ocicrypt.conf << EOF
+{
+    "key-providers": {
+        "attestation-agent": {
+            "grpc": "127.0.0.1:50000"
+        }
+    }
+}
+EOF
+```
+
+```shell
+OCICRYPT_KEYPROVIDER_CONFIG=ocicrypt.conf \
+nydusify convert --source docker.io/library/busybox:latest \
+--target [REGISTRY_URL]/busybox:encrypted-nydus \
+--encrypt-recipients provider:attestation-agent \
+--nydus-image /usr/bin/nydus-image
+```
+
+Be sure replace [REGISTRY_URL] with the desired registry URL like `docker.io/myregistry`.
+
+----
+
+## Deploy encrypted Nydus image as a CoCo workload on CC HW
+Here is a sample yaml for encrypted Nydus image deploying:
+
+```shell
+cat << EOT | tee encrypted-image-test-busybox.yaml
```

Comment:
I suggest leaving just the yaml content here, set the syntax-highlight type to `yaml`, and remove the `cat << EOT...` and `EOT`.

If you really want this as a shell command, use `cat > encrypted-image-test-busybox.yaml << EOT` as the first line (as we do above for `ocicrypt.conf`).